appveyor.yml

```diff
@@ -1,11 +1,65 @@
 image: Visual Studio 2015
 
+environment:
+  NUGET_API_KEY:
+    secure: Rh/UHYn7HGj8a7LFWo9C1rSA0sHSQNV3wIqj9bvwRR8yLmCXu9JmbNkmkctk2cBn
+  MYGET_API_KEY:
+    secure: hA4Ut1N2lrrdEtAN24Bty/FNiU0d/Ur/dLYSqpr8jKHOvoO7MU4jD+KwzUvATh+E
+
 pull_requests:
   do_not_increment_build_number: true
 
 build_script:
 - ps: |
-    & .\build.cmd CompleteBuild
+    # A trick to test if variable is defined. Number could never be null.
+    $isPullRequest = -not ($Env:APPVEYOR_PULL_REQUEST_NUMBER -eq $null)
+    $isTagBuild = $Env:APPVEYOR_REPO_TAG -eq "true"
+    $branchName = $Env:APPVEYOR_REPO_BRANCH
+
+    Write-Host "[BUILD CONTEXT INFO] Is PR: $isPullRequest, is tag: $isTagBuild, branch name: '$branchName'" -ForegroundColor Green
+
+    if($isTagBuild)
+    {
+      $tagName = $Env:APPVEYOR_REPO_TAG_NAME
+      $isSemVerTag = $tagName -match '^v.*'
+      # We consider version as alpha if it contains the "-alpha" pre-release tag.
+      $isAlphaVerTag = $tagName -match '^v[\d\.]+-alpha[\d\.]*$'
+
+      if(-not $isSemVerTag)
+      {
+        Write-Host "[Tag] Trigger build only - this is not a SemVer tag. Tag name: $tagName" -ForegroundColor Green
+        & .\build.cmd CompleteBuild BuildVersion=git BuildNumber=$($Env:APPVEYOR_BUILD_NUMBER)
+      }
+      # Alpha version is published to the private feed only as tag's purpose is to start the vNext versions count.
```

Comment:
It could happen that we decide to have Release Candidates, which are still on `vNext` branch, however should be published to the public feed. 
I want our configuration to allow such things happen. If don't need it - we will simply not create tags on vNext branch (except the `alpha` ones), however if we need that - we have that for free.